packages/react/src/tabs/tab/TabsTab.tsx

```diff
@@ -57,17 +50,11 @@ export const TabsTab = React.forwardRef(function Tab(
     // because the index is needed for Tab internals
   } = useCompositeItem<TabsTab.Metadata>({ metadata: tabMetadata });
 
-  const tabValue = valueProp ?? index;
-
   // the `selected` state isn't set on the server (it relies on effects to be calculated),
   // so we fall back to checking the `value` param with the selectedTabValue from the TabsContext
   const selected = React.useMemo(() => {
-    if (valueProp === undefined) {
-      return index < 0 ? false : index === selectedTabValue;
-    }
-
-    return valueProp === selectedTabValue;
-  }, [index, selectedTabValue, valueProp]);
+    return value === selectedTabValue;
+  }, [selectedTabValue, value]);
```

Comment:
Not sure, but I wonder if this makes the `isNavigatingRef.current` redundant? It fixes an issue with disabled tabs specifically due to how this value got calculated

Reply:
This doesn't need the `useMemo` wrapper anymore